eslint.config.mjs

```diff
@@ -0,0 +1,85 @@
+import typescriptEslint from "@typescript-eslint/eslint-plugin";
+import prettier from "eslint-plugin-prettier";
+import simpleImportSort from "eslint-plugin-simple-import-sort";
+import _import from "eslint-plugin-import";
+import { fixupPluginRules } from "@eslint/compat";
+import globals from "globals";
+import tsParser from "@typescript-eslint/parser";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import js from "@eslint/js";
+import { FlatCompat } from "@eslint/eslintrc";
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+const compat = new FlatCompat({
+    baseDirectory: __dirname,
+    recommendedConfig: js.configs.recommended,
+    allConfig: js.configs.all
+});
+
+export default [{
+    ignores: ["**/node_modules/", "tests/qunit/"],
+}, {
+    plugins: {
+        "@typescript-eslint": typescriptEslint,
+        prettier,
+        "simple-import-sort": simpleImportSort,
+        import: fixupPluginRules(_import),
+    },
+
+    languageOptions: {
+        globals: {
+            ...globals.browser,
+            ...globals.node,
+        },
+
+        parser: tsParser,
+        ecmaVersion: 5,
+        sourceType: "commonjs",
+
+        parserOptions: {
+            project: ["./tsconfig.json", "./tsconfig.tools.json"],
+        },
+    },
+
+    rules: {
+        "no-console": "warn",
+        "prettier/prettier": "warn",
+        "simple-import-sort/imports": "warn",
+        "simple-import-sort/exports": "warn",
+        "import/first": "error",
+        "import/no-duplicates": "error",
+        "import/newline-after-import": "warn",
+        camelcase: "warn",
+    },
+}, ...compat.extends(
+    "eslint:recommended",
+    "prettier",
+    "plugin:@typescript-eslint/eslint-recommended",
+    "plugin:@typescript-eslint/recommended",
+).map(config => ({
+    ...config,
+    files: ["**/*.ts"],
+})), {
+    files: ["**/*.ts"],
+
+    rules: {
+        "@typescript-eslint/no-inferrable-types": "off",
+
+        "simple-import-sort/imports": ["warn", {
+            groups: [["^.*/vex.*$"], ["^.*/index$"], ["^\\u0000"], ["^@?\\w"], ["^"], ["^\\."]],
```

Comment:
It looks like we lost some of the comments from the `.eslintrc` file.

Given that eslint configs are difficult to understand, can we restore all the comments to their appropriate places, so that a future person can decipher this configuration more easily?

Reply:
done